src/components/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,235 @@
+                        changeValue += 5;
+                        break;
+                    case elapsedSeconds > 5000:
+                        changeValue += 10;
+                        break;
+                    default:
+                        changeValue = 1;
+                }
+                handleCount(type, changeValue);
+                elapsedSeconds += 250;
+            }, 250);
+        };
+        getNewInputValue();
+    };
+
+    const handleKeyDown = (event: KeyboardEvent) => {
+        const { key, shiftKey } = event;
+        if (key === 'Backspace' || !isNaN(Number(key))) {
+            return;
+        }
+        event.preventDefault();
+        isShift.current = shiftKey;
+        if (DECREMENT_KEYS.includes(key)) {
+            handleCount(NumberInputIncrement.DECREMENT);
+        } else if (INCREMENT_KEYS.includes(key)) {
+            handleCount(NumberInputIncrement.INCREMENT);
+        }
+        if (onKeyDown) {
+            onKeyDown(event);
+        }
+    };
+
+    const handleKeyUp = (event: KeyboardEvent) => {
+        event.preventDefault();
+        isShift.current = event.shiftKey;
+        if (inputElement.current) {
+            handleOnChange(inputElement.current.value);
+        }
+    };
+
+    const handleCount = (type: NumberInputIncrement, changeValue = 1) => {
+        const getNewInputValue = () => {
+            const value = inputElement.current?.value;
+            const currentValue = Number(value) || 0;
+            changeValue = isShift.current ? stepInterval : changeValue;
+            switch (type) {
+                case NumberInputIncrement.DECREMENT:
+                    return currentValue - changeValue;
+                case NumberInputIncrement.INCREMENT:
+                    return currentValue + changeValue;
+                default:
+                    return currentValue;
+            }
+        };
+
+        const newValue = getNewInputValue();
+        setValue(newValue);
+    };
+
+    const setValue = (newValue: number) => {
+        if (inputElement.current) {
+            inputElement.current.value = newValue.toString();
+            if (isMouseHold.current) {
+                return;
+            }
+            handleOnChange(inputElement.current.value);
+        }
+    };
+
+    const handleOnChange = useCallback(
+        (value: string) => {
+            onChange?.(Number(value));
+        },
+        [onChange],
+    );
+
+    useEffect(() => {
+        clearInterval(timer.current);
+    }, []);
+
+    return (
+        <div
+            className={merge([
+                'tw-flex tw-items-center tw-h-9 tw-gap-2 tw-px-3 tw-transition tw-text-s tw-font-sans tw-relative tw-bg-white dark:tw-bg-transparent tw-border tw-rounded tw-line-strong hover:tw-line-x-strong focus-within:tw-line-xx-strong',
+                status ? validationClassMap[status] : '',
+            ])}
+            data-test-id={dataTestId}
+        >
+            {decorator ? (
```

Comment:
It's trivial as there is nothing wrong with the ternary but just to keep the same style everywhere.
```
 {decorator && (
              <div
                    className={merge([
                        'tw-flex tw-items-center tw-justify-center tw-pl-1',
                        disabled || readOnly ? 'tw-text-black-60' : 'tw-text-black-80',
                    ])}
                    data-test-id={`${dataTestId}-decorator`}
                >
                    {decorator}
                </div>
            )}

Reply:
@imoutaharik I had `x && y` originally and @SamuelAlev asked to use the `x ? y : z` format.

Reply:
I just find it. Fair enough.